subcommands/keys/ca_create.go

```diff
@@ -77,19 +75,20 @@ func getDeviceCaCommonName(factory string) string {
 }
 
 func doCreateCA(cmd *cobra.Command, args []string) {
+	var storage x509.KeyStorage
 	factory := viper.GetString("factory")
 	certsDir := args[0]
 	logrus.Debugf("Create CA for %s under %s", factory, certsDir)
 	subcommands.DieNotNil(os.Chdir(certsDir))
 
-	if len(hsmModule) > 0 {
-		if len(hsmPin) == 0 {
+	if len(hsm.Module) > 0 {
+		if len(hsm.Pin) == 0 {
 			fmt.Println("ERROR: --hsm-pin is required with --hsm-module")
 			os.Exit(1)
 		}
-		os.Setenv("HSM_MODULE", hsmModule)
-		os.Setenv("HSM_PIN", hsmPin)
-		os.Setenv("HSM_TOKEN_LABEL", hsmTokenLabel)
+		storage = &x509.KeyStorageHsm{Hsm: hsm}
+	} else {
+		storage = &x509.KeyStorageFiles{}
```

Comment:
I think we need to separate factory-root storage from online-ca storage. It doesn't make sense to have both keys on the same device. You want to keep the root keypair offline and in safe and the online keypair is going to wind up in some manufacturing location.

Reply:
I think the online device CA private key is stored on our backend. I don't think we need to store its public part/cert in HSM.

I guess the offline/local CA keypair is meant here.

It sounds like we need two params, one to specify "destination" for factory CA and another one for offline/local device CA.

Reply:
sorry, i meant offline-ca